### DISCUSSION TITLE
How to implement a paywall?

### DISCUSSION THREAD

**Discussion by o-simardcasanova-net** [quarto-dev/quarto-cli#3816] [Q&A]
Hi,

I'm assuming the answer to the question "how do you implement a paywall with Quarto?" is "you can't". Quarto is a really capable editor for technical and scientific documents. With the creator economy currently booming, the kind of niche content Quarto is designed for is extremely valuable — if there is a *simple* way to monetize it. Substack and Ghost and WordPress have some capabilities, but they are no match when it comes to citations, equations, cross-referencing, tables, and so on.

I'm assuming the answer to my question is "you can't" mostly because static files are fundamentally incapable to do this kind of things. But I'm still asking, just in case.

(To clarify: I'm not asking if in theory a paywall is possible, I'm asking if a simple-ish paywall solution exists.)

Thanks,
Olivier

**Comment by mcanouil** [comment]
Quarto does not have a Paywall implementation and I don't believe it ever will.

This being said, you can include whatever you want in your Quarto documents especially for `html` format such as JavaScript or anything.
So, you can learn to do that (there are good tutorial, code, etc. out-there) and use the `include-in-header` option to add it to the content you want to hide behind a paywall.

PS: could you change the category from "general" to "Q&A"? This way you can mark an answer as the answer, thus let others know it has been answered.

**Comment by jimjam-slam** [comment]
Since the payment and user processing needs to be done server-side, I think this is really something you need to look to a host to do.

Some of the static site hosts that work well with Quarto also include identity (user management and access) functionality. For example, [Netlify](https://netlify.com) has [Netlify Identity](https://docs.netlify.com/visitor-access/identity).

That doesn't get you all the way to pay-gated premium content on its own, but [here's a demo](https://github.com/stripe-samples/netlify-stripe-subscriptions) where someone set Netlify Identity up with Stripe for payment processing. There are a few other moving pieces (eg. a database), but there is [a tutorial linked](https://www.netlify.com/blog/2020/07/13/manage-subscriptions-and-protect-content-with-stripe/) to the demo that you could look into.

Once you have the identity part set up, you can [write redirects](https://docs.netlify.com/visitor-access/role-based-access-control/) to keep people out of the pages you want paywalled! So you could, for example, keep all your premium content within a folder in Quarto and then redirect people from that folder to a sign-up page if they don't have a "Paid subscriber" role 🙂 

(Sorry, I know this isn't very out-of-the-box!)

**Comment by SaremS** [comment]
I am currently working on a paywall for my own blog: https://github.com/SaremS/quarto_paywall
Not completely ready yet but maybe it helps.
